pkg/utils/values_patch.go

```diff
@@ -40,15 +41,47 @@ func (p *ValuesPatch) ToJsonPatch() (jsonpatch.Patch, error) {
 	return patch, nil
 }
 
-// Apply calls jsonpatch.Apply to mutate a JSON document according to the patch.
-func (p *ValuesPatch) Apply(doc []byte) ([]byte, error) {
+// ApplyStrict calls jsonpatch.Apply to transform a JSON document according to the patch.
+//
+// - "remove" operation errors are not ignored.
+// - absent paths are not ignored.
+func (p *ValuesPatch) ApplyStrict(doc []byte) ([]byte, error) {
 	patch, err := p.ToJsonPatch()
 	if err != nil {
 		return nil, err
 	}
 	return patch.Apply(doc)
 }
 
+// Apply calls jsonpatch.Apply to transform an input JSON document.
+//
+// - errors from "remove" operations are ignored.
+func (p *ValuesPatch) Apply(doc []byte) ([]byte, error) {
+	for _, op := range p.Operations {
+		patch, err := op.ToJsonPatch()
+		if err != nil {
+			return nil, err
+		}
+		newDoc, err := patch.Apply(doc)
+		// Ignore errors for remove operation.
+		if err != nil && op.Op == "remove" {
+			errStr := err.Error()
+			if strings.HasPrefix(errStr, "error in remove for path:") {
```

Comment:
Is this something we can rely on? I assume we check prefix here only to ignore `"remove operation failed to decode path"` error from the `op.Path` method.

Maybe we can validate paths of operations separately from the `patch.Apply` method and continue on every error here. WDYT?

Reply:
As discussed, prefixes are extracted as constants, and error checking is extracted to the function.